File: verify/tests/fixtures/ts-hexagonal/src/adapters/postgres-order-repo.ts
import { Order } from '../domain/order';
import { OrderRepository } from '../ports/order-repository';

export class PostgresOrderRepo implements OrderRepository {
    db: any;

    save(order: Order): void {
        // implementation
    }

    findById(id: string): Order {
        return new Order();
    }
}

File: test/fixtures/rust-clean-arch/src/infra/postgres_order_repo.rs
use crate::domain::order::Order;
use crate::port::order_repository::OrderRepository;

pub struct PostgresOrderRepo {
    pub db: PgPool,
}

impl PostgresOrderRepo {
    pub fn new(db: PgPool) -> Self {
        Self { db }
    }
}

impl OrderRepository for PostgresOrderRepo {
    fn save(&self, order: Order) -> Result<(), String> {
        Ok(())
    }

    fn find_by_id(&self, id: String) -> Result<Order, String> {
        todo!()
    }
}
